icegatherer.go

```diff
@@ -176,7 +174,10 @@ func (g *ICEGatherer) Gather() error {
 // Close prunes all local candidates, and closes the ports.
 func (g *ICEGatherer) Close() error {
 	g.lock.Lock()
-	defer g.lock.Unlock()
+	defer func() {
+		g.lock.Unlock()
+		g.setState(ICEGathererStateClosed)
```

Comment:
Before `setState(ICEGathererStateClosed)` only happened on a successful close, now it will happen during an error as well.

Reply:
Ah, I misread original behavor.
It bit complex to do it cleanly, but I'll fix it soon.

Reply:
I would just move it back for now, the rest of the PR looks 100% to me though!

Reply:
Fixed to call only after successful close.